File: Reto1.ts
export {};//El archivo se convierte en un módulo, y sus declaraciones ya no estarán en el ámbito global.

function prepareGifts(gifts: number[]): number[] {//parámetro gifts es un array de números : la función devuelve un array de números
    // Eliminar duplicados usando Set y convertir de nuevo a array
    const uniqueGifts = [...new Set(gifts)];
    // Ordenar la lista en orden ascendente
    uniqueGifts.sort((a, b) => a - b);
    return uniqueGifts;
}

// Ejemplos de uso
const gifts1: number[] = [3, 1, 2, 3, 4, 2, 5];
const preparedGifts1: number[] = prepareGifts(gifts1);
console.log(preparedGifts1); // [1, 2, 3, 4, 5]

const gifts2: number[] = [6, 5, 5, 5, 5];
const preparedGifts2: number[] = prepareGifts(gifts2);
console.log(preparedGifts2); // [5, 6]

const gifts3: number[] = [];
const preparedGifts3: number[] = prepareGifts(gifts3);
console.log(preparedGifts3); // []

File: Reto1.py
def prepare_gifts(gifts):
  unique_gifts = list(set(gifts))
  unique_gifts.sort()
  return unique_gifts

gifts1 = [3, 1, 2, 3, 4, 2, 5]
prepared_gifts1 = prepare_gifts(gifts1)
print(prepared_gifts1)

gifts2 = [6, 5, 5, 5, 5]
prepared_gifts2 = prepare_gifts(gifts2)
print(prepared_gifts2) 

gifts3 = []
prepared_gifts3 = prepare_gifts(gifts3)
print(prepared_gifts3)
